### PR TITLE
Extract exp table validation into separate function.

### PR DESCRIPTION
**Pull request by 4144**

### CODE REVIEW

File: src/map/pc.c [hunk 1]
```diff
@@ -10354,16 +10354,7 @@ int pc_readdb(void) {
 		}
 	}
 	fclose(fp);
-	for (i = 0; i < JOB_MAX; i++) {
-		if (!pc->db_checkid(i)) continue;
-		if (i == JOB_WEDDING || i == JOB_XMAS || i == JOB_SUMMER)
-			continue; //Classes that do not need exp tables.
-		j = pc->class2idx(i);
```
Comment (MishimaHaruna) [inline src/map/pc.c]: if I'm not mistaken, now that this line is removed, the `j` variable is unused in this function?

Comment (4144) [inline src/map/pc.c]: No, j variable used in other lines. For example in RENEWAL_DROP RENEWAL_EXP

Comment (MishimaHaruna) [inline src/map/pc.c]: You're certainly right, I can indeed see it now. Inspecting code through this web application isn't the same as checking the actual file on a proper text editor.

**Comment by MishimaHaruna** [comment]
Thank you!

**Pull request merged**